tests/TestRegistryTest.cpp

```diff
@@ -323,3 +323,38 @@ TEST(TestRegistry, ResetPluginsWorks)
     myRegistry->resetPlugins();
     LONGS_EQUAL(0, myRegistry->countPlugins());
 }
+
+TEST(TestRegistry, listGroupNames)
+{
+    test1->setGroupName("GROUP_1");
+    myRegistry->addTest(test1);
+    test2->setGroupName("GROUP_2");
+    myRegistry->addTest(test2);
+    test3->setGroupName("GROUP_1");
+    myRegistry->addTest(test3);
+
+    myRegistry->listTestGroupNames(*result);
+    SimpleString s = output->getOutput();
+    STRCMP_CONTAINS("GROUP_1 ", s.asCharString());
+    STRCMP_CONTAINS("GROUP_2 ", s.asCharString());
+    LONGS_EQUAL(1, s.count("GROUP_1 "));
```

Comment:
@mmidor I agree. Otherwise, one would have to employ TEST_ORDERED() or rely on the unspecified (backward) ordering of tests, which is bad. There is no good enough reason to use TEST_ORDERED() in this case.